### PR TITLE
feat: introduce nyc weather features to feature-engineering on fabric single-tech sample

### PR DESCRIPTION
**Pull request by thurstonchen**
# Type of PR

- Documentation changes
- Code changes

## Purpose

Introduce NYC weather dataset.
- to make our model training scenario a bit closer to the practical use case :- )
- switch to LightGBMRegressor model to have higher model performance metrics.
- now we create two feature sets, nyctaxi and nycweather, the latter could be reused by other model training requirement, which is a good showcase on why we need feature store.

Besides introducing NYC weather data of year 2022, we also add Jan. 2023 taxi trip and weather data to do the batch inferencing. 
Hi @promisinganuj, if you agree this PR, we'll share those new data files to you via Teams, thanks! :- )  

## Does this introduce a breaking change? If yes, details on what can break
NO

## Author pre-publish checklist

- [ ] Added test to prove my fix is effective or new feature works
- [ ] No PII in logs
- [x] Made corresponding changes to the documentation

## Validation steps

- Put new NYC weather files to the public storage account.
- Run the Fabric data pipeline again.
- Run model training and inferencing notebooks, now the latter notebook also requires attaching the same lakehouse used by other notebooks. 

## Issues Closed or Referenced
N/A

### CODE REVIEW

**Comment by promisinganuj** [comment]
Hi @siliang-j-1225 as discussed, please review this PR in the next two weeks. I will be incorporating your review comments. Thanks for your help.

CC: @devlace

**Comment by promisinganuj** [comment]
HI @siliang-j-1225 , Hope you are well. Just checking whether you have bandwidth to look at it this week.

**Comment by siliang-j-1225** [comment]
> HI @siliang-j-1225 , Hope you are well. Just checking whether you have bandwidth to look at it this week.

Hey @promisinganuj, yes I will work on it this week.

File: single_tech_samples/fabric/feature_engineering_on_fabric/README.md [hunk 1]
```diff
@@ -55,11 +55,13 @@ In addition to the main flow, there are optional steps for performing 'explorato
...
 
 Base URL: [https://stmdwpublic.blob.core.windows.net/](https://stmdwpublic.blob.core.windows.net/)
 
-This subset contains data for the year 2022, and each month is available as a separate parquet file. The data includes anonymized travel details like departure points, destinations, times, distances, and costs. The data, in conjunction with taxi zone maps and lookup tables, aids in various research fields such as identifying frequent pickup and drop-off zones in the city.
+This subset of the yellow taxi trip dataset contains data from Jan. 2022 to Jan. 2023, and each month is available as a separate parquet file. The data includes anonymized travel details like departure points, destinations, times, distances, and costs. The data, in conjunction with taxi zone maps and lookup tables, aids in various research fields such as identifying frequent pickup and drop-off zones in the city.
```
Comment (siliang-j-1225) [inline single_tech_samples/fabric/feature_engineering_on_fabric/README.md:62]:
```suggestion
The yellow taxi trip dataset spans from Jan. 2022 to Jan. 2023, providing data for each month in distinct parquet files. It encompasses anonymized information about travel details, including departure points, destinations, times, distances, and costs. In conjunction with taxi zone maps and lookup tables, this dataset supports research in areas like identifying frequent pickup and drop-off zones in the city.
```

**Review by siliang-j-1225 — APPROVED** [review]

**Pull request merged**